crux-http-server/src/crux/http_server/entity.clj

```diff
@@ -128,61 +149,67 @@
                                                         :body [:div.error-box not-found-message]
                                                         :options opts
                                                         :results {:entity-results
                                                                   {"error" not-found-message}}}))
-                           error (let [error-message (.getMessage ^Exception error)]
-                                   (util/raw-html {:title "/entity"
-                                                   :body [:div.error-box error-message]
-                                                   :options opts
-                                                   :results {:entity-results
-                                                             {"error" error-message}}}))
+                           cause (util/raw-html {:title "/_crux/entity"
+                                                 :body [:div.error-box cause]
+                                                 :options opts
+                                                 :results {:entity-results
+                                                           {"error" cause}}})
                            entity-history (try
                                             (util/raw-html {:body (entity-history->html res)
-                                                            :title "/entity?history=true"
+                                                            :title "/_crux/entity?history=true"
                                                             :options opts
                                                             :results {:entity-results (iterator-seq entity-history)}})
                                             (finally
                                               (.close entity-history)))
                            :else (util/raw-html {:body (entity->html res)
-                                                 :title "/entity"
+                                                 :title "/_crux/entity"
                                                  :options opts
                                                  :results {:entity-results entity}}))]
         (.getBytes resp ^String charset)))))
 
 (defn ->edn-encoder [_]
   (reify
     mfc/EncodeToOutputStream
-    (encode-to-output-stream [_ {:keys [entity entity-history] :as res} _]
+    (encode-to-output-stream [_ {:keys [entity ^Cursor entity-history] :as res} _]
       (fn [^OutputStream output-stream]
         (with-open [w (io/writer output-stream)]
-          (if entity-history
-            (try
-              (print-method (iterator-seq entity-history) w)
-              (finally
-                (cio/try-close entity-history)))
-            (print-method entity w)))))))
+          (cond
+            entity-history (try
+                             (if (.hasNext entity-history)
+                               (print-method (iterator-seq entity-history) w)
```

Comment:
might be clearer to go with `(print-method (or (iterator-seq entity-history) '()) w) ` rather than this `if`?